src/billing/Project.ts

```diff
@@ -38,32 +54,47 @@ const groupByBillingAccountStatus = (billingProject, workspaces) => {
   //   W is the number of workspaces in a billing project (can be very large for GP).
   // Note we need to perform this search W times for each billing project; using a set reduces time
   // complexity by an order of magnitude.
-  return _.mapValues((ws) => new Set(ws), _.groupBy(group, workspaces));
+  return _.mapValues(
+    (workspacesInGroup) => new Set<WorkspaceInfo>(workspacesInGroup),
+    _.groupBy(group, workspaces) as Record<BillingAccountStatus, WorkspaceInfo[]>
+  ) as Record<BillingAccountStatus, Set<WorkspaceInfo>>;
 };
 
 const spendReportKey = 'spend report';
 
-const ProjectDetail = ({
-  authorizeAndLoadAccounts,
-  billingAccounts,
-  billingProject,
-  isOwner,
-  reloadBillingProject,
-  workspaces,
-  refreshWorkspaces,
-}) => {
+interface ProjectDetailProps {
+  authorizeAndLoadAccounts: () => Promise<void>;
+  billingAccounts: Record<string, GoogleBillingAccount>;
+  billingProject: BillingProject;
+  isOwner: boolean;
+  reloadBillingProject: () => Promise<unknown>;
+  workspaces: WorkspaceWrapper[];
+  refreshWorkspaces: () => Promise<void>;
+}
+
+const ProjectDetail = (props: ProjectDetailProps): ReactNode => {
+  const {
+    authorizeAndLoadAccounts,
+    billingAccounts,
+    billingProject,
+    isOwner,
+    reloadBillingProject,
+    workspaces,
+    refreshWorkspaces,
+  } = props;
+
   // State
   const { query } = Nav.useRoute();
   // Rather than using a localized StateHistory store here, we use the existing `workspaceStore` value (via the `useWorkspaces` hook)
 
   const [projectUsers, setProjectUsers] = useState(() => StateHistory.get().projectUsers || []);
-  const projectOwners = _.filter(_.flow(_.get('roles'), _.includes(billingRoles.owner)), projectUsers);
+  const projectOwners: User[] = _.filter(_.flow(_.get('roles'), _.includes(billingRoles.owner)), projectUsers);
 
   const [updating, setUpdating] = useState(false);
   const [showBillingModal, setShowBillingModal] = useState(false);
   const [tab, setTab] = useState(query.tab || 'workspaces');
 
-  const workspacesInProject = useMemo(
+  const workspacesInProject: WorkspaceInfo[] = useMemo(
```

Comment:
Again, adding the types allowed me to understand that this `_.map` usage was pulling the `workspace` variable out of the workspaces….